lib/plug/conn.ex

```diff
@@ -446,13 +446,13 @@ defmodule Plug.Conn do
   `:sent`.
   """
   @spec put_resp_header(t, binary, binary) :: t
-  def put_resp_header(%Conn{resp_headers: headers, state: state} = conn, key, value) when
-      is_binary(key) and is_binary(value) and state != :sent do
-    %{conn | resp_headers: List.keystore(headers, key, 0, {key, value})}
+  def put_resp_header(%Conn{state: :sent}, _key, _value) do
```

Comment:
We should check for this rather: https://github.com/lexmag/plug/blob/sent-state/lib/plug/conn.ex#L175

Can you please change update_resp_header too?